pkg/operator/invalidcertscontroller/invalid_certs_controller.go

```diff
@@ -0,0 +1,145 @@
+package invalidcertscontroller
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/prometheus/common/model"
+	"io/ioutil"
+	"k8s.io/client-go/transport"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	prometheusapi "github.com/prometheus/client_golang/api"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+type invalidCertsController struct {
+	operatorClient          v1helpers.OperatorClient
+	prometheusQuery         string
```

Comment:
not sure if leaving this as an arbitrary string is a great idea. however, the underlying prometheus queries might vary a lot.

Reply:
I'm hesitant on this approach, what kind of metric consumers are meant to look after? I only know the one from kube-apiserver, shouldn't that controller be defined only there? If you're thinking about other consumers I'm not sure if they have that metric available, I was rather considering this particular controller to look over pre-defined certificates and checking for their correctness which is what that kube-apiserver metric does, no?

Reply:
this is also supposed to be used in cluster-authentication-operator and a yet-to-be-introduced metric in oauth-server, see https://github.com/openshift/enhancements/blob/9225dbdfa07d7479f412e6d2c5f275f484def5b0/enhancements/authentication/invalid-certs.md#oauth-server

Reply:
Additionally, we can replace the metric controller in etcd-operator, see https://github.com/openshift/cluster-etcd-operator/tree/master/pkg/operator/metriccontroller.

Reply:
Gotcha, that makes sens, thx :)